feat: add architecture doc (#10)
strimzi/strimzi-ui: client/Panel/README.md

> # Panel

Reviewer: Can we change this to Panels

Author: I actually started with this (and the other types) as plural, and changed them to singular! Will wait for a pass from @nictownsend but happy to have them either way.

Reviewer 2: Yep I agree with plurals for all folders. The root is the collection (so plural) - and each child is implicitly a singular instance.

Reviewer 2: But refer to `Panel` and equivalent when talking about an instance (like below)

Author: Thank you both - will update to re puralise in the next commit

> A `Panel` component represents a top level element of a UI, such as an entire page, or significant capability of a page. A `Panel`s responsibility is not only to compose `Element` and `Group` components to implement the required UI, but to own and manage activities such as data fetching, state reduction, integration with the wider UI and so on. For example, a `Panel` would on mount make a request for and own the response for a piece of data, and while this request is happening, swap in and out `Group` and or `Element` components to represent the loading/error/success states.

Reviewer: Wondering if we need to have a separate directory for Pages instead of grouping it with Panel.

Author: I think the component implementation will remain a `Panel`, but the useage to make them a `Page` may move to a `Pages` directory. Part of the work I am doing under #12 highlighted this to me as well - doing that keeps things seperate and makes usage easier too. Will complete my prototype/write up findings/reflect them here

> A `Panel` component represents a top level element of a UI, such as an entire page, or significant capability of a page. A `Panel`s responsibility is not only to compose `Element` and `Group` components to implement the required UI, but to own and manage activities such as data fetching, state reduction, integration with the wider UI and so on. For example, a `Panel` would on mount make a request for and own the response for a piece of data, and while this request is happening, swap in and out `Group` and or `Element` components to represent the loading/error/success states.

Reviewer 2: ```suggestion
A `Panel` component represents a top level element of a UI - the primary/significant capability of a page. A `Panel`s responsibility is not only to compose `Element` and `Group` components to implement the required UI, but to own and manage activities such as data fetching, state reduction, integration with the wider UI and so on. For example, a `Panel` would on mount make a request for and own the response for a piece of data, and while this request is happening, swap in and out `Group` and or `Element` components to represent the loading/error/success states.
```

Then elsewhere we should make it clear that a `Page` is what the user sees (i.e, navigation + content).

Author: As mentioned above, this is going to need a tweak given how my Navigation work #12 has progressed. I will update to reflect the outcome/difference between a Panel component, and ultimatley what a user sees.

## Test approach

To be defined in https://github.com/strimzi/strimzi-ui/issues/7

## Expected files

For a given Panel component `Foo`, the expected files are as follows:

```
Panel/
  index.js

>   Foo/

Reviewer 2: ```suggestion
  ViewTopics/
```

    index.js
    README.md
    View.js
    Model.js
    Styling.scss
```

Where:

> - index.js acts as a barrel file, exporting all public elements of this component/the components in the Panel directory

Reviewer 2: ```suggestion
- index.js acts as a barrel file, exporting the public API of this component/component bundle.
```
